### ISSUE TITLE
`check-script-has-no-table-name` incorrectly flags partially Jinja-templated CTE names as raw table references

### ISSUE THREAD

**Issue by bzly** [dbt-checkpoint/dbt-checkpoint#350]
**Describe the bug**
When a CTE is defined using a Jinja variable (e.g. `cte_{{ var }}`), `check-script-has-no-table-name` fails to recognise it as a CTE alias and reports the text before the Jinja block as a hardcoded table reference (e.g. `cte_`).

**To Reproduce**
1. Create a dbt model with a Jinja-templated CTE name:

```sql
{% set foo = 'bar' %}

WITH

cte_{{ foo }} AS (
    SELECT * FROM {{ ref('baz') }}
)

SELECT * FROM cte_{{ foo }}
```

2. Run `check-script-has-no-table-name` against the file.
3. Observe:

```
i/have/noname/but/imustscream.sql: does not use source() or ref() macros for tables:
 - cte_
```

**Expected behavior**
The test should not fail on any CTE aliases.

**Version:**
v2.0.8

Additional context
n/a

**Comment by k0x2e** [comment]
Closing as duplicate of #58. The Jinja-templated CTE name (`cte_{{ foo }} AS (...)`) is another variant of the same CTE alias detection failure tracked in #58. The tokeniser strips Jinja blocks, leaving `cte_` as the apparent token — which isn't matched against the CTE alias `}}` that was registered. Tracking the fix in #58.
